internal/handlers/common/params.go

```diff
@@ -147,6 +154,48 @@ func GetWholeNumberParam(value any) (int64, error) {
 	}
 }
 
+// GetLimitStageParam returns $limit stage argument from the provided value.
+// It returns the proper error if value doesn't meet requirements.
+func GetLimitStageParam(value any) (int64, error) {
```

Comment:
I added `GetLimitParam` in #2254; let's merge that other PR first, then see if it is possible to re-use it

Reply:
`GetLimitStageParam` already handles errors to match message. Also from my understanding, `$limit` stage could have different requirements, for example `find`'s limit can be 0, but `$limit` cannot.

Reply:
I don't think we should maintain two separate functions that do almost the same job. From your description, it sounds like we could move some checks to `GetLimitParam`, call it from `GetLimitStageParam` and add a check for 0.

Reply:
If I understand our final agreement, we stay with separate function for a while and find a better way later, right?

Reply:
That's fine for now